lemmy_api/src/report.rs

```diff
@@ -0,0 +1,300 @@
+use actix_web::web::Data;
+use std::str::FromStr;
+
+use lemmy_db::{comment_report::*, comment_view::*, post_report::*, post_view::*, Reportable, ReportType,};
+use lemmy_structs::{blocking, report::*};
+use lemmy_utils::{APIError, ConnectionId, LemmyError};
+use lemmy_websocket::{LemmyContext, UserOperation, messages::SendUserRoomMessage};
+
+use crate::{check_community_ban, get_user_from_jwt, is_mod_or_admin, Perform};
+
+const MAX_REPORT_LEN: usize = 1000;
+
+#[async_trait::async_trait(?Send)]
+impl Perform for CreateReport {
```

Comment:
I mention this below, but this should be split out into createcommentreport, and createpostreport.